lava/magma/runtime/runtime.py

```diff
@@ -273,8 +275,6 @@ def join(self):
             port.join()
         for port in self.runtime_to_service_data:
             port.join()
-        for actor in self._actors:
-            actor.join()
 
     @property
     def global_time(self):
```

Comment:
I cannot comment on set_var/get_var below but there is actually a few minor and bigger issues that I just noticed now:

What's the purpose of this:
`runtime_srv_id: int = ev.runtime_srv_id`
We do not really save any characters by introducing this local variable. 

Same here:
`buffer_shape: ty.Tuple[int, ...] = buffer.shape`
buffer_shape is no shorter than buffer.shape

This makes no sense:
`if idx:
                buffer = buffer[idx]`
If the user provides no idx, then we can assume that value is dense with shape Var.shape.
If the user provides an idx, then the expectation is that the user has already provided a subset of the values as of the Var. In this case idx just specifies which elements in the original Var the values in 'value' correspond to.
Furthermore the whole purpose of providing 'idx' is to send the idx along to the RuntimeSrv to presumable send (or receive in case of get) less data and for the RuntimeSrv and Process to know which values to set. If you just use 'idx' to extract some values for 'value' and then send this reduced 'value' vector along, how will the receiver know at which elements in the destination vector to change?

Also, all of this this
`buffer_shape: ty.Tuple[int, ...] = buffer.shape
num_items: int = np.prod(buffer_shape).item()
buffer = buffer.reshape((1, num_items))`
can be replaced by numpy.ndarray.flatten(..)

Finally, this is likely going to be very slow:
`for i in range(num_items):
    data_port.send(enum_to_np(buffer[0, i]))`
Can we not send the entire vector?